lib/rules/no-meta-schema-default.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+const { getStaticValue } = require('@eslint-community/eslint-utils');
+const utils = require('../utils');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'disallow rules `meta.schema` properties to include defaults',
+      category: 'Rules',
+      recommended: false,
+      url: 'https://github.com/eslint-community/eslint-plugin-eslint-plugin/tree/HEAD/docs/rules/no-meta-schema-default.md',
+    },
+    schema: [],
+    messages: {
+      foundDefault: 'Disallowed default value in schema.',
+    },
+  },
+
+  create(context) {
+    const sourceCode = context.sourceCode || context.getSourceCode(); // TODO: just use context.sourceCode when dropping eslint < v9
+    const { scopeManager } = sourceCode;
+    const ruleInfo = utils.getRuleInfo(sourceCode);
+    if (!ruleInfo) {
+      return {};
+    }
+
+    const schemaNode = utils.getMetaSchemaNode(ruleInfo.meta, scopeManager);
+    if (!schemaNode) {
+      return {};
+    }
+
+    const schemaProperty = utils.getMetaSchemaNodeProperty(
+      schemaNode,
+      scopeManager,
+    );
+
+    if (schemaProperty?.type === 'ObjectExpression') {
+      checkSchemaElement(schemaProperty, true);
+    } else if (schemaProperty?.type === 'ArrayExpression') {
+      for (const element of schemaProperty.elements) {
+        checkSchemaElement(element, true);
+      }
+    }
+
+    return {};
+
+    function checkSchemaElement(node) {
+      if (node.type !== 'ObjectExpression') {
+        return;
+      }
+
+      for (const { key, value } of node.properties) {
+        if (!key) {
+          continue;
+        }
```

Comment:
nits: It might be clearer to compare "type" directly (I'm assuming that the intention here is to skip the `...obj`):

```suggestion
      for (const { type, key, value } of node.properties) {
        if (type !== 'Property') {
          continue;
        }
```